From d8466aca83b11ab8b0870e775536306782d73e9f Mon Sep 17 00:00:00 2001
Message-Id: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
From: Gajanan Bhat <gajanan.bhat@intel.com>
Date: Thu, 29 Jan 2015 19:36:03 +0530
Subject: [PATCH 01/28] MUST_REBASE [VPG]: drivers/video/adf: Add more color
 formats supported by primary plane

Primary plane supports DRM color formats DRM_FORMAT_XRGB1555 and
DRM_FORMAT_ARGB1555. Add these as supported formats.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5534
Change-Id: I55aec98b218e64c60b35f78ab57995db687599af
Signed-off-by: Gajanan Bhat <gajanan.bhat@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_pri_plane.c |   14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_pri_plane.c b/drivers/video/adf/intel/core/vlv/vlv_pri_plane.c
index e7536f5..c0f9070 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_pri_plane.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_pri_plane.c
@@ -38,6 +38,18 @@ static const struct format_info format_mapping[] = {
 	},
 
 	{
+		.drm_format = DRM_FORMAT_XRGB1555,
+		.hw_config = DISPPLANE_BGRX555,
+		.bpp = 2,
+	},
+
+	{
+		.drm_format = DRM_FORMAT_ARGB1555,
+		.hw_config = DISPPLANE_BGRA555,
+		.bpp = 2,
+	},
+
+	{
 		.drm_format = DRM_FORMAT_RGB565,
 		.hw_config = DISPPLANE_BGRX565,
 		.bpp = 2,
@@ -94,6 +106,8 @@ static const struct format_info format_mapping[] = {
 
 static const u32 pri_supported_formats[] = {
 	DRM_FORMAT_C8,
+	DRM_FORMAT_XRGB1555,
+	DRM_FORMAT_ARGB1555,
 	DRM_FORMAT_RGB565,
 	DRM_FORMAT_XRGB8888,
 	DRM_FORMAT_ARGB8888,
-- 
1.7.9.5

